src/sagemaker/predictor.py

```diff
@@ -157,6 +157,106 @@ def _create_request_args(self, data, initial_args=None, target_model=None, targe
+                This is required if ``instance_type``, ``accelerator_type``, or ``model_name`` is
+                specified. Otherwise, the values from the existing endpoint configuration's
+                ProductionVariants are used.
+            instance_type (str): The EC2 instance type to deploy the endpoint to.
+                This is required if ``initial_instance_count`` or ``accelerator_type`` is specified.
+                Otherwise, the values from the existing endpoint configuration's
+                ``ProductionVariants`` are used.
+            accelerator_type (str): The type of Elastic Inference accelerator to attach to
+                the endpoint, e.g. "ml.eia1.medium". If not specified, and
+                ``initial_instance_count``, ``instance_type``, and ``model_name`` are also ``None``,
+                the values from the existing endpoint configuration's ``ProductionVariants`` are
+                used. Otherwise, no Elastic Inference accelerator is attached to the endpoint.
+            model_name (str): The name of the model to be associated with the endpoint.
+                This is required if ``initial_instance_count``, ``instance_type``, or
+                ``accelerator_type`` is specified and if there is more than one model associated
+                with the endpoint. Otherwise, the existing model for the endpoint is used.
+            tags (list[dict[str, str]]): The list of tags to add to the endpoint
+                config. If not specified, the tags of the existing endpoint configuration are used.
+                If any of the existing tags are reserved AWS ones (i.e. begin with "aws"),
+                they are not carried over to the new endpoint configuration.
+            kms_key (str): The KMS key that is used to encrypt the data on the storage volume
+                attached to the instance hosting the endpoint If not specified,
+                the KMS key of the existing endpoint configuration is used.
+            data_capture_config_dict (dict): The endpoint data capture configuration
+                for use with Amazon SageMaker Model Monitoring. If not specified,
+                the data capture configuration of the existing endpoint configuration is used.
+
+        Raises:
+            ValueError: If there is not enough information to create a new ``ProductionVariant``:
+
+                - If ``initial_instance_count``, ``accelerator_type``, or ``model_name`` is
+                  specified, but ``instance_type`` is ``None``.
+                - If ``initial_instance_count``, ``instance_type``, or ``accelerator_type`` is
+                  specified and either ``model_name`` is ``None`` or there are multiple models
+                  associated with the endpoint.
+        """
+        production_variants = None
+
+        if initial_instance_count or instance_type or accelerator_type or model_name:
```

Comment:
I’m just curious about the None check here. Why are we checking the args and raise ValueError instead of making them required variables?

Reply:
my thinking was that if someone wanted to update only the tags, KMS key, or data capture config, they shouldn't need to provide all the other info too.